src/main.cpp

```diff
@@ -0,0 +1,69 @@
+#include <stdio.h>
+#include <vector>
+#include <iostream>
+
+#define ADD 		'+'
+#define SUBSTRACT	'-'
+#define MULTIPLY	'*'
+#define DIVIDE		'/'
+
+//TODO: Dynamic typing, for now every value is int
+#define PseudoValue int // To enable dynamic typing we need to use one type for every value
+
+#define Program std::vector<ASTNode*>
+
+struct Runtime {
+	PseudoValue acc;
+	void error(char* err) {
+		std::cerr << "Unexpected error has occured:\n" << err << std::endl;
+		std::exit(1);
+	}
+};
+
+
+struct ASTNode {
+	PseudoValue value;
+	void (*resolver)(Runtime* r, ASTNode* self);
+	ASTNode* left;
+	ASTNode* right;
+	void resolve(Runtime* r) {
+		resolver(r, this);
+	}
+};
+
+void constIntResolver(Runtime* r, ASTNode* self) {
+	r->acc = self->value;
+}
+
+ASTNode* createConstInt(PseudoValue value) {
+	return new ASTNode{value, &constIntResolver, nullptr, nullptr};
+}
+
+void sumIntResolver(Runtime* r, ASTNode* self) {
```

Comment:
same with these. If we don't want these methods to be in struct think it'd be better to create a separate class for them could be even static methods if needed.